### ISSUE TITLE
Guzzle getBody with dynamic clients throws error

### ISSUE THREAD

**Issue by randomsymbols** [spawnia/sailor#85]
Guzzle version 7+ cannot get body after it has already been accessed https://github.com/guzzle/guzzle/issues/2452
When using https://github.com/spawnia/sailor#dynamic-clients the client's response can be accessed (for example by Guzzle middleware https://docs.guzzlephp.org/en/stable/handlers-and-middleware.html) and it breaks the Sailor implementation of Guzzle response with `getBody` with this error:

```
An uncaught Exception was encountered

Type: Spawnia\Sailor\Error\InvalidDataException

Message: Received a response that is invalid JSON:

Filename: /var/www/html/vendor/spawnia/sailor/src/Response.php

Line Number: 59

Backtrace:

File: /var/www/html/vendor/spawnia/sailor/src/Response.php
Line: 43
Function: fromJson

File: /var/www/html/vendor/spawnia/sailor/src/Client/Guzzle.php
Line: 33
Function: fromResponseInterface

File: /var/www/html/vendor/spawnia/sailor/src/Operation.php
Line: 87
Function: request

File: /var/www/html/vendor/spawnia/sailor/src/Operation.php
Line: 59
Function: fetchResponse 
```

One solution may be to use `__toString` instead of `getBody`

**Comment by spawnia** [comment]
> When using [spawnia/sailor#dynamic-clients](https://github.com/spawnia/sailor#dynamic-clients) the client's response can be accessed

Let's make sure we don't confuse terminology here. I would take client to mean *an implementation of `\Spawnia\Sailor\Client`*. The responsibility of implementing classes is to fulfill this method signature:

```php
public function request(string $query, \stdClass $variables = null): Response;
```

It is entirely up to you to implement this in a way that does not break.

Concerning `\Spawnia\Sailor\Response::fromResponseInterface()` - it is programmed against `\Psr\Http\Message\ResponseInterface`, which does not have a `__toString()` method. If you decide to consume a message beforehand by calling its `getBody()` method, you must create a new message to pass along.

In conclusion, I think there is nothing that can or should be changed in this library, and you have the power to fix the problem you brought upon yourself.
